### DISCUSSION TITLE
Ingest data error

### DISCUSSION THREAD

**Discussion by disturbed-mystic1** [mayooear/ai-pdf-chatbot-langchain#47] [Q&A]
![image](https://user-images.githubusercontent.com/68948344/227504233-371532ba-b982-4880-8b4a-a92036fe5de0.png)

I am getting this error. I checked my openai usage and still have plenty of $ left. Also, I am using us central as the pinecone environment. 
Also, I did all the troubleshooting steps mentioned in the discussion. 

![image](https://user-images.githubusercontent.com/68948344/227504697-c07d4f54-1cb3-4270-981b-7a44a0e5a2a7.png)
Also there seems to be some error with the number of arguments

**Comment by gasper94** [comment]
I followed this steps:
1. clone the repo & cd into it
2. pnpm install dependencies
3.  create .env and add your keys
![1](https://user-images.githubusercontent.com/18272791/227692506-23c15217-905b-4cba-a459-f5e7312cfbfe.png)

4. Make sure you add Pinecone information on Pinecone.ts (Index name & name space)
![2](https://user-images.githubusercontent.com/18272791/227693054-da3ab890-f8ee-48d3-a0fc-6a3964153892.png)
![3](https://user-images.githubusercontent.com/18272791/227693301-63b279c2-ae0a-44eb-84c3-2372b68c01ed.png)

Note: make sure you set the index dimension to 1536

5. run "npm run ingest"

**Reply by mayooear** [reply]
Thank you for the visuals. I'll add them to the README.

**Comment by mayooear** [comment]
> ![image](https://user-images.githubusercontent.com/68948344/227504233-371532ba-b982-4880-8b4a-a92036fe5de0.png)
> 
> I am getting this error. I checked my openai usage and still have plenty of $ left. Also, I am using us central as the pinecone environment. Also, I did all the troubleshooting steps mentioned in the discussion.
> 
> ![image](https://user-images.githubusercontent.com/68948344/227504697-c07d4f54-1cb3-4270-981b-7a44a0e5a2a7.png) Also there seems to be some error with the number of arguments

it looks like you upgraded LangChain (and potentially Pinecone) to the latest version. You have to downgrade to the version used in this repo.

**Reply by douyujie** [reply]
I have exactly the same version with this repo but got the same issue

**Reply by douyujie** [reply]
btw how to switch the environment in pinecone to us-east1-gcp? There is no option to do this

**Comment by disturbed-mystic1** [comment]
Need a premium version for this unfortunately

**Reply by douyujie** [reply]
Oh thanks! Have you tried in other env to confirm it was the culprit? My env is us-west4-gcp

**Reply by disturbed-mystic1** [reply]
Nope like mentioned by the creator of this project, the langchain version can cause issues.

**Comment by disturbed-mystic1** [comment]
It works now I just redid from start using us central as environment. My
guess is something was wrong with my first index so I created a new one.
